core/src/main/java/org/hibernate/ogm/type/impl/EnumType.java

```diff
@@ -87,7 +87,8 @@ public Object nullSafeGet(Tuple rs, String name, SharedSessionContractImplemento
 
 	private void initEnumValues() {
 		if ( enumValues == null ) {
-			this.enumValues = coreEnumType.returnedClass().getEnumConstants();
+			Class<Enum<?>> returnedClass = coreEnumType.returnedClass();
+			this.enumValues = returnedClass.getEnumConstants();
```

Comment:
I can't see a difference between the 2 codes. But either is OK to me.

Reply:
I think it's a leftover

Reply:
So, as a quick experiment, I removed this commit and I have a compilation error but I don't see any difference with the classes involved. I think I'm too tired now, if you want to try to solve this Java puzzle, be my guest :)

Reply:
Otherwise I will check tomorrow

Reply:
It might be some weird generics issue with the compiler. It's not worth wasting time IMHO. Let's keep your new code.